examples/catalyst_contrastive/barlow_twins.py

```diff
@@ -0,0 +1,169 @@
+
+
+class CustomRunner(dl.Runner):
+    def handle_batch(self, batch) -> None:
+        if self.is_train_loader:
+            (pos_1, pos_2), targets = batch
+            feature_1, out_1 = self.model(pos_1)
+            _, out_2 = self.model(pos_2)
+            self.batch = {
+                "embeddings": feature_1,
+                "out_1": out_1,
+                "out_2": out_2,
+                "targets": targets,
+            }
+        else:
+            images, targets = batch
+            features, _ = self.model(images)
+            self.batch = {"embeddings": features, "targets": targets}
+
+
+parser = argparse.ArgumentParser(description="Train Barlow Twins on cifar-10")
+parser.add_argument("--feature_dim", default=128, type=int, help="Feature dim for latent vector")
+parser.add_argument("--temperature", default=0.5, type=float, help="Temperature used in softmax")
+parser.add_argument(
+    "--batch_size", default=512, type=int, help="Number of images in each mini-batch"
+)
+parser.add_argument(
+    "--epochs", default=1000, type=int, help="Number of sweeps over the dataset to train"
+)
+parser.add_argument(
+    "--num_workers", default=8, type=float, help="Number of workers to process a dataloader"
+)
+parser.add_argument(
+    "--offdig_lambda",
+    default=0.005,
+    type=float,
+    help="Lambda that controls the on- and off-diagonal terms",
+)
+parser.add_argument(
+    "--logdir", default="./logdir", type=str, help="Logs directory (tensorboard, weights, etc)",
+)
+if __name__ == "__main__":
+
+    # args parse
+    args = parser.parse_args()
+
+    # hyperparams
+    feature_dim, temperature = args.feature_dim, args.temperature
+    offdig_lambda = args.offdig_lambda
+    batch_size, epochs, num_workers = args.batch_size, args.epochs, args.num_workers
+
+    # data
+    train_data = torchvision.datasets.CIFAR10(
+        root="data", train=True, transform=CifarPairTransform(train_transform=True), download=True
+    )
+    test_data = torchvision.datasets.CIFAR10(
+        root="data",
+        train=False,
+        transform=CifarPairTransform(train_transform=False, pair_transform=False),
+        download=True,
+    )
+
+    train_loader = DataLoader(train_data, batch_size=batch_size, shuffle=True, pin_memory=True)
+    valid_loader = DataLoader(test_data, batch_size=batch_size, pin_memory=True)
+
+    callbacks = [
+        dl.ControlFlowCallback(
+            dl.CriterionCallback(input_key="out_1", target_key="out_2", metric_key="loss"),
+            loaders="train",
+        ),
+        dl.SklearnClassifierCallback(
+            feature_key="embeddings",
+            target_key="targets",
+            train_loader="train",
+            valid_loader="valid",
+            sklearn_classifier_fn=LogisticRegression,
+            sklearn_metric_fn=partial(top_k_accuracy_score, **{"k": 1}),
+        ),
+        dl.OptimizerCallback(metric_key="loss"),
+    ]
+
+    model = Model(feature_dim, arch="resnet50").cuda()
```

Comment:
```suggestion
    model = Model(feature_dim, arch="resnet50")
```
should be good enough ;)